### PR TITLE
Add sealed secret chart

### PR DESCRIPTION
**Pull request by olib963**
#### What this PR does / why we need it:

Provides a helm chart for the [sealed-secrets](https://github.com/bitnami-labs/sealed-secrets) project.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by unguiculus** [comment]
Thanks for the PR. A few things from quickly looking oever it:

* Please compare with what you get using `helm create` and update accordingly.
* Please follow our RBAC best practices:
  https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by olib963** [comment]
Thank you for the feedback, I will get to work on that this Friday :D

File: stable/sealed-secrets/templates/NOTES.txt
Comment (anguslees) [inline stable/sealed-secrets/templates/NOTES.txt]:
```suggestion
# note the use of `--dry-run` - this does not create a secret in your cluster
```

File: stable/sealed-secrets/templates/NOTES.txt
Comment (anguslees) [inline stable/sealed-secrets/templates/NOTES.txt:30]: I suggest adding a mention here somewhere that the SealedSecret and Secret must use the same namespace/name (it's a security check to stop different users in the same cluster decrypting each other's SealedSecrets).  This is the # 1 confusion people encounter when using sealed-secrets for the first time.

**Review by anguslees — COMMENTED** [review]
(I am the upstream author)
I will leave the helm-specific bits up to other people, but fwiw the manifest looks ok to me.

**Comment by olib963** [comment]
@prydonius @unguiculus I believe I have updated to follow the RBAC best practices and have added `resouorces` etc. that are generated by `helm create`. I hope this is what you were looking for.

File: stable/sealed-secrets/templates/deployment.yaml
Comment (alexmt) [inline stable/sealed-secrets/templates/deployment.yaml]: Thank you for doing it @olib963 ! Unless I'm missing something obvious image should be hardcoded to `quay.io/bitnami/sealed-secrets-controller`, is not it?

Comment (olib963) [reply stable/sealed-secrets/templates/deployment.yaml]: You are absolutely correct, find and replace error. Fixed it.

**Comment by charlex** [comment]
This seems like it would be a very good addition to helm. Any update on this status?

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml]: Use `apps/v1`

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml]: Add `release´ label to selector.

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml:11]: Add a `spec.selector`.

File: stable/sealed-secrets/templates/service-account.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/service-account.yaml]: `{{ if .Values.serviceAccount.create }}`

Should not depend on `rbac.create`

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml]: Remove `namespace: {{ .Release.Namespace }}` 
 on all resources. It's unnecessary.

File: stable/sealed-secrets/values.yaml
Comment (unguiculus) [inline stable/sealed-secrets/values.yaml]: Leave default empty

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml]: `quay.io/bitnami/sealed-secrets-controller` should be configurable as `image.repository`.

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [inline stable/sealed-secrets/templates/deployment.yaml]: I'd suggest to consistent with the list style using this:

```yaml
args: 
  - --key-name
  - {{ .Values.secretName }}
```

File: stable/sealed-secrets/values.yaml
Comment (unguiculus) [inline stable/sealed-secrets/values.yaml]: Should be 

```yaml
rbac:
  create: true
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/sealed-secrets/templates/deployment.yaml
Comment (olib963) [inline stable/sealed-secrets/templates/deployment.yaml:22]: @unguiculus Is this okay to leave here? I didn't want to enforce the use of `serviceAccountName` in the case you aren't using RBAC resources because it is unlikely you will need the service account at all. Is this an acceptable practice?

**Comment by olib963** [comment]
@unguiculus I hope this doesn't come across as impatient, just wondering if we would be able to get this merged in some time soon? Was planning on adding it to my terraform, just need to know if I should crack on with a manual implementation for now. Cheers :D

**Comment by olib963** [comment]
@unguiculus Anything I need to do to get this merged in or someone else you know I could assign? Sorry I am new to this repo.

**Comment by johnraz** [comment]
I'd be very interested into having this merged as well.
Do you have plans to propose it as part of https://github.com/helm/charts ?

File: stable/sealed-secrets/README.md
Comment (unguiculus) [inline stable/sealed-secrets/README.md]: `>= 1.9`

File: stable/sealed-secrets/README.md
Comment (unguiculus) [inline stable/sealed-secrets/README.md]: Remove comment

File: stable/sealed-secrets/templates/_helpers.tpl
Comment (unguiculus) [inline stable/sealed-secrets/templates/_helpers.tpl]: Should be `{{ default "default" .Values.serviceAccount.name }}`

File: stable/sealed-secrets/templates/deployment.yaml
Comment (unguiculus) [reply stable/sealed-secrets/templates/deployment.yaml:22]: Remove the `if` statement. If you consider my other comment, the default service account is used if nothing is configured.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
I'd suggest you update the chart to use the new label conventions before we merge it:
https://github.com/helm/helm/blob/master/docs/chart_best_practices/labels.md

File: stable/sealed-secrets/templates/sealedsecret-crd.yaml
Comment (stefanprodan) [inline stable/sealed-secrets/templates/sealedsecret-crd.yaml:3]: I would add:

```yaml
  annotations:
    "helm.sh/resource-policy": keep
```

This way you don't lose the secrets if you delete the sealed secret release.

File: stable/sealed-secrets/Chart.yaml
Comment (unguiculus) [inline stable/sealed-secrets/Chart.yaml:7]: A chart needs to have maintainers. Since you're adding the chart, this should be you. Are you willing to maintain the chart and handle future issues and PRs for it?

File: stable/sealed-secrets/Chart.yaml
Comment (unguiculus) [inline stable/sealed-secrets/Chart.yaml]: `">=1.9.0-0"`

File: stable/sealed-secrets/Chart.yaml
Comment (unguiculus) [inline stable/sealed-secrets/Chart.yaml]: Start with `1.0.0`

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/sealed-secrets/Chart.yaml
Comment (stefanprodan) [reply stable/sealed-secrets/Chart.yaml:7]: @olib963 I volunteer as a maintainer if you want, here is my info:

```yaml
maintainers:
  - name: stefanprodan
    email: stefan.prodan@gmail.com
```

You should also have a OWNERS file, here is an example: https://github.com/helm/charts/blob/master/stable/weave-cloud/OWNERS

Comment (olib963) [reply stable/sealed-secrets/Chart.yaml:7]: Thank you very much :D

**Comment by unguiculus** [comment]
Retesting alone won't help. The chart created a CRD which is still around. I'll have to clean this up.

**Comment by olib963** [comment]
@unguiculus Sorry i didn't realise it was using an existing cluster not a real existing one. Would you suggest making the `keep` configurable or just completely removing it for now?

**Comment by unguiculus** [comment]
The chart should clean up after itself. At least for CI, you should make this configurable.

See https://github.com/helm/charts/tree/master/test#providing-custom-test-values

**Comment by olib963** [comment]
@unguiculus Thank you so much for your help and feedback :D

**Review by unguiculus — APPROVED** [review]

**Pull request merged**
